src/api/platform/models/api_user_key.rs

```diff
@@ -0,0 +1,53 @@
+use serde::Deserialize;
+/// User Key struct handed to us by the API
+#[derive(Clone, Debug, Deserialize)]
+pub struct ApiUserKey {
+    id: ApiKeyId,
+    name: String,
+    user_id: UserId,
+    api_access: bool,
+    #[serde(rename = "pem")]
+    public_key_pem: String,
+    fingerprint: String,
+    created_at: String,
+}
+
+impl ApiUserKey {
+    /// Key Id
+    pub fn id(&self) -> &ApiKeyId {
+        &self.id
+    }
+
+    /// Name of the Key
+    pub fn name(&self) -> &String {
+        &self.name
+    }
+
+    /// User Id of the Owner of the Key
+    pub fn user_id(&self) -> &str {
+        &self.user_id
+    }
```

Comment:
While this works, it should return a reference to the type. If the UserId type changes to something more concrete like a Uuid this will result in an error